src/multiconfigwidget.cpp

```diff
@@ -1,119 +1,211 @@
 /*
     SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
-    SPDX-FileCopyrightText: Milian Wolff <milian.wolff@kdab.com>
-    SPDX-FileCopyrightText: 2016-2022 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    SPDX-FileCopyrightText: 2022-2023 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
 
     SPDX-License-Identifier: GPL-2.0-or-later
 */
 
 #include "multiconfigwidget.h"
 
+#include <algorithm>
+
 #include <QComboBox>
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QPushButton>
 
+#include <KUrlRequester>
+
+#include "ui_multiconfigwidget.h"
+
+namespace {
+template<typename Widget, class RestoreFunc>
+void restoreWidget(QWidget* parent, const KConfigGroup& group, RestoreFunc restoreFunc)
+{
+    Q_ASSERT(parent);
+    Q_ASSERT(group.isValid());
+    for (auto* child : parent->findChildren<Widget*>()) {
```

Comment:
this looks super fishy and brittle, please make the restoration logic static and explicit, instead of dynamic like proposed here

this could otherwise break if e.g. we save/restore data within complex widgets that use e.g. a QLineEdit internaly _and_ give it an object name

meaning: this logic here is fine, as long as the list of widgets that should be restored is hardcoded and does not rely on `findChildren`. I.e. change the API such that we pass along the list of widgets, or register them somehow, maybe just pass it along as a `QVector<QWidget*>` to the `setChildWidget` API

Reply:
would it be fine if the search only goes one level deep? I am not really keen on writing a save/restore function for every form